sdk/monitor/azure-monitor-opentelemetry-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorStatsbeatTest.java

```diff
@@ -136,13 +138,18 @@ private static void generateMetric(OpenTelemetry openTelemetry) {
                 AttributeKey.stringKey("name"), "apple", AttributeKey.stringKey("color"), "red"));
     }
 
-    private void verifyStatsbeatTelemetry(CustomValidationPolicy customValidationPolicy) {
+    private void verifyStatsbeatTelemetry(CustomValidationPolicy customValidationPolicy) throws IOException {
         TelemetryItem attachStatsbeat =
             customValidationPolicy.getActualTelemetryItems().stream()
                 .filter(item -> item.getName().equals("Statsbeat"))
                 .filter(item -> {
-                    MetricsData metricsData = (MetricsData) item.getData().getBaseData();
-                    return metricsData.getMetrics().stream().allMatch(metricDataPoint -> metricDataPoint.getName().equals("Attach"));
+                    MetricsData metricsData = null;
+                    try {
+                        metricsData = toMetricsData(item.getData().getBaseData());
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
```

Comment:
maybe we should just change `toMetricsData` (and friends) to catch IOException and rethrow RuntimeException? then wouldn't need to handle here